Makefile

```diff
@@ -0,0 +1,65 @@
+# Copyright (C) 2019 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+CRC_VERSION = 0.0.1
+COMMIT_SHA=$(shell git rev-parse --short HEAD)
+
+# Go and compilation related variables
+BUILD_DIR ?= out
+
+GOOS ?= $(shell go env GOOS)
+GOARCH ?= $(shell go env GOARCH)
+ORG := github.com/code-ready
+REPOPATH ?= $(ORG)/crc
+ifeq ($(GOOS),windows)
+	IS_EXE := .exe
+endif
+CRC_BINARY ?= $(CURDIR)/bin/crc$(IS_EXE)
+TIMEOUT ?= 10800s
+PACKAGES := go list ./... | grep -v /vendor | grep -v /out
+SOURCE_DIRS = cmd pkg
+
+
+# Linker flags
+VERSION_VARIABLES := -X $(REPOPATH)/pkg/version.crcVersion=$(CRC_VERSION) \
+	-X $(REPOPATH)/pkg/version.commitSha=$(COMMIT_SHA)
+
+LDFLAGS := $(VERSION_VARIABLES)
+
+# Start of the actual build targets
+
+.PHONY: $(CURDIR)/bin/crc$(IS_EXE)
+$(CURDIR)/bin/crc$(IS_EXE): $(ADDON_ASSET_FILE) ## Builds the binary into $CURDIR/bin
+	go install -tags "$(BUILD_TAGS_SYSTEMTRAY)" -pkgdir=$(ADDON_BINDATA_DIR) -ldflags="$(VERSION_VARIABLES)" ./cmd/crc
+
+$(BUILD_DIR)/$(GOOS)-$(GOARCH):
+	mkdir -p $(BUILD_DIR)/$(GOOS)-$(GOARCH)
+
+$(BUILD_DIR)/darwin-amd64/crc: $(BUILD_DIR)/$(GOOS)-$(GOARCH) ## Cross compiles the darwin executable and places it in $(BUILD_DIR)/darwin-amd64/crc
+	GOARCH=amd64 GOOS=darwin go build -ldflags="$(LDFLAGS)" -o $(BUILD_DIR)/darwin-amd64/crc ./cmd/crc
+
+$(BUILD_DIR)/linux-amd64/crc: $(ADDON_ASSET_FILE) $(BUILD_DIR)/$(GOOS)-$(GOARCH) ## Cross compiles the linux executable and places it in $(BUILD_DIR)/linux-amd64/crc
```

Comment:
`ADDON_ASSET_FILE` is not used variable as of now so we should remove it from here, in future if we need something like this, we can add it back.

Reply:
@praveenkumar right, I missed it.